VidaToDo/Features/Todos/Todo Creation Form/Interface/TodoFormViewController.swift

```diff
@@ -5,64 +5,58 @@
 import UIKit
 import VidaUIKit
 
-class TodoFormViewController: UIViewController, Routable {
-    
-    private let formFields = FormFields()
-
-    let viewModel = FormViewModel()
 
-    var closeButton = UIButton()
-    var addButton = UIButton()
-
-    private let disposeBag = DisposeBag()
+class TodoFormViewController: UIViewController {
+    // UI
+    private let closeButton = UIButton()
+    private let addButton = UIButton()
+    private let formFields = FormFields()
 
-    static func makeWithURL(_ screenURL: GlobalScreenURL) -> UIViewController? {
-        guard case .todoForm = screenURL else {
-            fatalLog("Invalid URL passed to view controller: \(self)")
-            return nil
-        }
-        return TodoFormViewController()
-    }
+    // private
+    private let viewModel = FormViewModel()
+    private let bag = DisposeBag()
 
+    // MARK: Config
     override func viewDidLoad() {
         super.viewDidLoad()
         view.backgroundColor = .white
         
         view.addSubview(formFields)
         formFields.left(10).top(100).right(10)
 
-        viewModel.bind(title: formFields.title, due: formFields.due, priority: formFields.priority)
-
-        viewModel.isValid?.subscribe(onNext: { [weak self] isValid in
-            self?.setButton(isEnabled: isValid)
-        }).disposed(by: disposeBag)
-
-        viewModel.hasSubmitted?.subscribe(onNext: { [weak self] isValid in
-            self?.dismiss()
-        }).disposed(by: disposeBag)
-
         closeButton.setTitle("X", for: .normal)
-        closeButton.addTarget(self, action: #selector(closeButtonClicked), for: .touchUpInside)
         closeButton.setTitleColor(.black, for: .normal)
         view.addSubview(closeButton)
 
         closeButton.right().top(12)
 
         addButton.setTitle("Add", for: .normal)
-        addButton.addTarget(self, action: #selector(addButtonClicked), for: .touchUpInside)
         view.addSubview(addButton)
 
         addButton.right().left().bottom()
+
+        subscribeTheViewModel(viewModel)
+        subscribeToViewModel(viewModel)
     }
 
-    @objc func closeButtonClicked() {
-        dismiss()
+    func subscribeTheViewModel(_ viewModel: FormViewModel) {
```

Comment:
I would have a function in the model called `runSubscriptions` I think. Then do `model.runSubscriptions`.

Reply:
A view shouldn't tell the view model to do anything, it just informs it. It is the view model's role to decide when to start emitting events on an observable.

Reply:
Telling the view model what to do is exactly what you're doing. See every call below.